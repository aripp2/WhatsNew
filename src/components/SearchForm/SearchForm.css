form {
  height: 75px;
  width: 100%;
  display: flex;
  justify-content: space-between;
  flex-direction: row;
  align-items: center;
}

input {
  height: 40px;
  width: 40%;
  font-size: 1.2em;
  margin-left: 25px;
  background-color: #141527;
  border: none;
}

.search-btn {
  height: 40px;
  width: 20%;
  color: #141527;
  font-size: 1.2em;
  margin-right: 5%;
  border-radius: 5px;
}