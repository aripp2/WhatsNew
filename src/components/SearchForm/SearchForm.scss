@import './src/variables.scss';


form {
  height: 75px;
  width: 100%;
  display: flex;
  justify-content: space-between;
  flex-direction: row;
  align-items: center;
}

input {
  height: 40px;
  width: 40%;
  font-size: 1.2em;
  margin-left: 25px;
  background-color: $dark-color;
  border: none;
}

.search-btn {
  height: 40px;
  width: 20%;
  color: $dark-color;
  font-size: 1.2em;
  margin-right: 5%;
  border-radius: 5px;
  background-color: $btn-color;

  &:hover {
    background-color: $highlight-color;
  }
}