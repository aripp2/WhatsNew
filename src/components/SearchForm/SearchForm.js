import React, { Component } from 'react';
import './SearchForm.css';

class SearchForm extends Component {
  constructor() {
    super();
    this.state = {
      searchInput: ''
    }
  }

render() {
  return (
    <form>
      <input
        className="search-input" 
        type="text"
        placeholder="Search for news article here."
        name="searchInput"
        
      />
      <button className="search-btn">Search Now</button>
    </form>

      )
  }
}

export default SearchForm;