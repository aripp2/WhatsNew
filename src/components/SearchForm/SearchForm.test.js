import React from 'react';
import { shallow } from 'enzyme';
import SearchForm from './SearchForm';

describe('SearchForm', () => {

  let wrapper;
  const mockSearchNews = jest.fn();
  const mockInputEvent = {
    target: {
      name: 'searchInput',
      value: 'dog'
    }
  };
  const mockEvent = { preventDefault: jest.fn() };

  beforeEach(() => {
    wrapper = shallow(<SearchForm
      searchNews={mockSearchNews}
     />);
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
  
  it('should update state when input changes', () => {
    wrapper.instance().handleChange(mockInputEvent);
    expect(wrapper.state('searchInput')).toEqual('dog');
  });

  it('should reset state when resetInput is called', () => {
    wrapper.instance().handleChange(mockInputEvent);
    expect(wrapper.state('searchInput')).toEqual('dog');
    wrapper.instance().resetInput();
    expect(wrapper.state('searchInput')).toEqual('');
  });

  it('should call searchNews and resetInput when submitSearch is called', () => {
    wrapper.instance().resetInput = jest.fn();
    wrapper.instance().submitSearch(mockEvent);
    expect(mockSearchNews).toHaveBeenCalled();
    expect(wrapper.instance().resetInput).toHaveBeenCalled();
  });

  it('should call submitSearch when button is clicked', () => {
    wrapper.instance().submitSearch = jest.fn();
    wrapper.instance().forceUpdate();
    wrapper.find('button').simulate('click', mockEvent);
    expect(wrapper.instance().submitSearch).toHaveBeenCalledWith(mockEvent);
  });

});