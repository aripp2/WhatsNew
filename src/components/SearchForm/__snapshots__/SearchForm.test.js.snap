// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match the snapshot 1`] = `
<form>
  <input
    className="search-input"
    name="searchInput"
    onChange={[Function]}
    placeholder="Search for news article here."
    type="text"
    value=""
  />
  <button
    className="search-btn"
    onClick={[Function]}
  >
    Search Now
  </button>
</form>
`;
